### PR TITLE
Add Inbound Auth config handler to expose OAuthAdmin Services

### PR DESCRIPTION
**Pull request by sahandilshan**
### Proposed changes in this pull request
- Add protocol service to expose OAuthAdmin Services

With this PR, we will expose the OAuthProtocol management services that are available on OAuthAdminServiceImpl with a separate protocol management service. This is to provide the capability to enable and disable audit logs. 

### When should this PR be merged
- This PR has to be merged after merging https://github.com/wso2/carbon-identity-framework/pull/5146

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthProtocolManagementService.java [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                    .registerAndRetrieveOAuthApplicationData(consumerApp, enableAuditing);
+            return createInboundAuthRequestConfig(createdOAuthApp);
+        } catch (IdentityOAuthAdminException e) {
+            if (e instanceof IdentityOAuthClientException) {
+                throw new IdentityApplicationManagementClientException(e.getMessage(), e);
+            } else if (e instanceof IdentityOAuthServerException) {
+                throw new IdentityApplicationManagementServerException(e.getMessage(), e);
+            } else {
+                throw new IdentityApplicationManagementException("Error while creating OAuth2 application", e);
+            }
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthProtocolManagementService.java:91]: why not have different catch blocks for each type?

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 2]
```diff
@@ -266,13 +266,30 @@ public void registerOAuthApplicationData(OAuthConsumerAppDTO application) throws
...
+    /**
+     * Same as {@link #registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO)} but with an option to disable
+     * the audit logs. This is to avoid logging duplicate logs.
+     *
+     * @param application    <code>OAuthConsumerAppDTO</code> with application information.
+     * @param enableAuditing Enable auditing or not.
+     * @return OAuthConsumerAppDTO Created OAuth application details.
+     * @throws IdentityOAuthAdminException Error when persisting the application information to the persistence store.
+     */
+    OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO application, boolean enableAuditing)
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:283]: Yes, this is intended. This method should be accessible by the InboundProtocol Service and should not be accessbile by other classes (Cannot be publically accesble). Hence this is declared as a package private

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/OAuthProtocolManagementServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
+package org.wso2.carbon.identity.oauth;
```
Comment (sahandilshan) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/OAuthProtocolManagementServiceTest.java:1]: Add the license and check other places

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OauthInboundAuthConfigHandler.java [hunk 4]
```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OauthInboundAuthConfigHandler.java:2]: 2024

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 5]
```diff
@@ -1357,7 +1406,7 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
                     + consumerKey);
         }
         handleInternalTokenRevocation(consumerKey, properties);
-        if (ApplicationMgtUtil.isLegacyAuditLogsDisabledInAppMgt()) {
+        if (enableAuditing && isEnableV2AuditLogs()) {
```
[the remark this answers is not in the document]
Comment (VivekVinushanth) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1409]: We have to log audits only when the global configuration **enableV2AuditLogs** and the service level **enableAuditing** flag is enabled.

[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:1409]: > why do have two conditions?

To publish a log, two conditions have to be true
1. Someone (Some other service) should directly call the admin service method
2. The user should have enabled publishing the latest audit logs (V2 Audit logs)

`enableAuditing` will check the first condition. If the admin service calls from the framework (from rest API flow), the enableAuding will be false and not publish the audit log from the protocol level. If the admin service calls from any other place, then the enableAuditing will be true, and logs will get published.

`enableV2AuditLogs` will check for the second condition where it checks whether the user has added the v2 audit log configurations on deployment.toml

**Review by VivekVinushanth — APPROVED** [review]

**Pull request merged**